include/swift/AST/Types.h

```diff
@@ -2295,7 +2323,8 @@ getSILFunctionLanguage(SILFunctionTypeRepresentation rep) {
 /// be 'thin', indicating that a function value has no capture context and can be
 /// represented at the binary level as a single function pointer.
 class AnyFunctionType : public TypeBase {
-  const Type Input;
+  const ArrayRef<CallArgParam> Input;
```

Comment:
I'd prefer to tail-allocate the parameters in the subclasses, so we don't pay for this extra pointer in such a common type.